core/src/main/java/com/datastax/oss/driver/internal/core/channel/ChannelFactory.java

```diff
@@ -99,11 +113,12 @@ public void setProtocolVersion(ProtocolVersion newVersion) {
       isNegotiating = true;
     }
 
-    connect(address, options, currentVersion, isNegotiating, attemptedVersions, resultFuture);
+    connect(node, address, options, currentVersion, isNegotiating, attemptedVersions, resultFuture);
     return resultFuture;
   }
 
   private void connect(
+      Node node,
```

Comment:
Do we need both `node` and `address`? Could we just use node all around? Also I saw that except for tests, the `ChannelFactory.connect(SocketAddress address, options)` was only used once and the method is called with `channelFactory.connect(node.getConnectAddress(), ..)` in https://github.com/datastax/java-driver/blob/4.x/core/src/main/java/com/datastax/oss/driver/internal/core/pool/ChannelPool.java#L278 , so maybe we can also remove the `ChannelFactory.connect(SocketAddress address, options)` now too?

Reply:
Oh I missed the "previous method retained for unit testing" in the initial comment. Not sure it is worth to keep this public method out there just for unit testing, but since it's the internal package, I guess it's debatable. Not a strong opinion but also as mentioned above the private method could maybe be simplified too.

Reply:
There might be a way for me to adjust the existing unit tests that use an address to use `Node` instead.   The problem was that they used a `LocalAddress` instead of `SocketAddress` so I couldn't simply wrap it in a mock `Node`, i'll look into it more though.

Reply:
Hrmm, looking at it again, I think this is a decent compromise.  `connect(SocketAddress address, DriverChannelOptions options)` uses default visibility (not public) and the class is in the internal package structure.  The tests that use `LocalAddress` use it for a reason so I think it may be better to leave them that way.

Reply:
Good point I hadn't seen that the visibility of `connect(SocketAddress, options)` had changed from public to package-local, sgtm for that then.